specification/common/attribute-precedence.md

```diff
@@ -0,0 +1,175 @@
+# Attribute Precedence on transformation to non-OTLP formats
+
+**Status**: [Experimental](../document-status.md)
+
+<details>
+<summary>Table of Contents</summary>
+
+<!-- toc -->
+
+- [Overview](#overview)
+- [Attribute hierarchy in the OTLP message](#attribute-hierarchy-in-the-otlp-message)
+- [Precedence per Signal](#precedence-per-signal)
+  - [Traces](#traces)
+    - [Span Events](#span-events)
+    - [Span links](#span-links)
+  - [Metrics](#metrics)
+    - [Metric exemplars](#metric-exemplars)
+  - [Logs](#logs)
+- [Considerations](#considerations)
+- [Example](#example)
+- [Useful links](#useful-links)
+
+<!-- tocstop -->
+
+</details>
+
+## Overview
+
+This document provides supplementary guidelines for the attribute precedence 
+that exporters should follow when translating from the hierarchical OTLP format
+to non-hierarchical formats.
+
+A mapping is required when flattening out attributes from the structured OTLP
+format, which has attributes at different levels (e.g., Resource attributes, 
+InstrumentationScope attributes, attributes on Spans/Metrics/Logs) to a
+non-hierarchical representation (e.g., OpenMetrics labels).
+In the case of OpenMetrics, the set of labels is completely flat and must have 
+unique labels only 
+(https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#labelset).
+Since OpenTelemetry allows for different levels of attributes, it is feasible
+that the same attribute appears multiple times on different levels.
+
+This document aims to provide guidance on how OpenTelemetry attributes can be 
+consistently mapped to flat sets.
+
+## Attribute hierarchy in OTLP messages
+
+Since the OTLP format is a hierarchical format, there is an inherent order in 
+the attributes.
+In this document, 
+[Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md)
+attributes are considered to be at the top of the hierarchy, since they are the
+most general attributes. 
+Attributes on individual Spans/Metric data points/Logs are at the bottom of the
+hierarchy, as they are most specialized and only apply to a subset of all data.
+
+**A more specialized attribute that shares an attribute key with more general 
+attribute will take precedence.** 
+
+In some cases it might be desirable to overwrite an attribute like this.
+<!-- TODO example -->
+
+When de-normalizing an OTLP message to a flat set of key-value pairs,
+attributes that are present on the Resource and InstrumentationScope levels will
+be duplicated for each Span/Metric data point/Log.
+
+## Precedence per Signal
+
+Below, the precedence for each of the signals is spelled out explicitly.
+Only Spans, Metric data points and LogRecords are considered.
```

Comment:
The casing seems a bit inconsistent here. Seems to reference directly `LogRecord` from the proto, but not the Metric equivalent. :thinking:

Reply:
Its hard, because the `Metric` has a oneof data and there are multiple choices (Gauge, Sum, Histogram,     ExponentialHistogram, Summary). Some of them have their own DataPoint specializations, so I would have to spell it out for each explicitly (or I am just vague).

Reply:
Maybe then just going with `spans, metric data points and log records` ?

Reply:
I changed it. I am reluctant still, since Span, Metric, and LogRecord are the names of the concepts in the proto - maybe `Span`s `Metric` data points and `LogRecord`s would work?

Reply:
I think we should not use the concepts from the proto and stick with "just words". Like they do in the glossary: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#signals

Reply:
I agree with Joao. This is not a blocking comment for me though. So feel free to ignore this. :slightly_smiling_face:

Reply:
Done, thanks!